contracts/core/OTCWrapper.sol

```diff
@@ -0,0 +1,690 @@
+import {SafeMath} from "@openzeppelin/contracts/utils/math/SafeMath.sol";
+import {ERC2771ContextUpgradeable} from "@openzeppelin/contracts-upgradeable/metatx/ERC2771ContextUpgradeable.sol";
+import {ContextUpgradeable} from "@openzeppelin/contracts-upgradeable/utils/ContextUpgradeable.sol";
+import {IERC20Permit} from "@openzeppelin/contracts/token/ERC20/extensions/draft-IERC20Permit.sol";
+import {MarginRequirementsWrapperInterface} from "../interfaces/otcWrapperInterfaces/MarginRequirementsWrapperInterface.sol";
+import {ControllerWrapperInterface} from "../interfaces/otcWrapperInterfaces/ControllerWrapperInterface.sol";
+import {AddressBookWrapperInterface} from "../interfaces/otcWrapperInterfaces/AddressBookWrapperInterface.sol";
+import {WhitelistWrapperInterface} from "../interfaces/otcWrapperInterfaces/WhitelistWrapperInterface.sol";
+import {UtilsWrapperInterface} from "../interfaces/otcWrapperInterfaces/UtilsWrapperInterface.sol";
+import {OracleWrapperInterface} from "../interfaces/otcWrapperInterfaces/OracleWrapperInterface.sol";
+import {IOtokenFactoryWrapperInterface} from "../interfaces/otcWrapperInterfaces/IOtokenFactoryWrapperInterface.sol";
+import {MinimalForwarder} from "@openzeppelin/contracts/metatx/MinimalForwarder.sol";
+import {SupportsNonCompliantERC20} from "../libs/SupportsNonCompliantERC20.sol";
+
+/**
+ * @title OTC Wrapper
+ * @author Ribbon Team
+ * @notice Contract that overlays Gamma Protocol for the OTC related interactions
+ */
+contract OTCWrapper is Initializable, OwnableUpgradeable, ReentrancyGuardUpgradeable, ERC2771ContextUpgradeable {
+    using SafeERC20 for IERC20;
+    using SupportsNonCompliantERC20 for IERC20;
+    using SafeMath for uint256;
+
+    AddressBookWrapperInterface public addressbook;
+    MarginRequirementsWrapperInterface public marginRequirements;
+    ControllerWrapperInterface public controller;
+    OracleWrapperInterface public oracle;
+    WhitelistWrapperInterface public whitelist;
+
+    /************************************************
+     *  EVENTS
+     ***********************************************/
+
+    /// @notice emits an event when an order is placed
+    event OrderPlaced(
+        uint256 indexed orderID,
+        address indexed underlyingAsset,
+        bool isPut,
+        uint256 strikePrice,
+        uint256 expiry,
+        uint256 premium,
+        uint256 notional,
+        address indexed buyer
+    );
+
+    /// @notice emits an event when an order is canceled
+    event OrderCancelled(uint256 orderID);
+
+    /// @notice emits an event when an order is executed
+    event OrderExecuted(
+        uint256 indexed orderID,
+        address collateralAsset,
+        uint256 premium,
+        address indexed seller,
+        uint256 indexed vaultID,
+        address oToken,
+        uint256 initialMargin
+    );
+
+    /// @notice emits an event when collateral is deposited
+    event CollateralDeposited(uint256 indexed orderID, uint256 amount);
```

Comment:
lets add `address acct` to collateral deposited event

Reply:
https://github.com/ribbon-finance/GammaProtocol-OTC/pull/2/commits/cd9059a905d57e4e46743313750d3f090268864c